bazel: openroad fails to link currently due to some missing symbols in slang elab

Openroad currently fails to link to due to what looks like some slang dependency not linking everything needed in third-party/ ? @oharboe or @povik maybe you have an idea ?

It is strange that it links in the CI apparently, so there might be some ambiguity that makes it accidentally work sometimes.

```
$ bazel build -c opt src/... -k 
INFO: Invocation ID: f7498947-808c-4006-9305-02c5061a7ee1
DEBUG: /home/hzeller/src/my/OpenROAD/bazel/notification.bzl:11:14: 
================================================================================
  NOTIFICATION: Use --config=opt to build with LTO (Link Time Optimization)
                and get better performance at the expense of longer build time.
================================================================================
INFO: Analyzed 2850 targets (114 packages loaded, 24927 targets configured).
ERROR: /home/hzeller/src/my/OpenROAD/BUILD.bazel:213:10: Linking openroad [for tool] failed: (Exit 1): clang failed: error executing CppLink command (from target //:openroad) /nix/store/pdmwvbf5aimvs5ji8wgkgiv88c33k6xq-clang-wrapper-22.1.5/bin/clang @bazel-out/k8-opt-exec-ST-d57f47055a04/bin/openroad-0.params

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::bitstream_member_offset(slang::ast::FieldSymbol const&):(.text._ZN14slang_frontend23bitstream_member_offsetERKN5slang3ast11FieldSymbolE+0x9f): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Symbol const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::bitstream_member_offset(slang::ast::FieldSymbol const&):(.text._ZN14slang_frontend23bitstream_member_offsetERKN5slang3ast11FieldSymbolE+0xba): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Symbol const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::inside_comparison(slang_frontend::EvalContext&, ir::Value, slang::ast::Expression const&):(.text._ZN14slang_frontend17inside_comparisonERNS_11EvalContextEN2ir5ValueERKN5slang3ast10ExpressionE+0x647): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Expression const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::inside_comparison(slang_frontend::EvalContext&, ir::Value, slang::ast::Expression const&):(.text._ZN14slang_frontend17inside_comparisonERNS_11EvalContextEN2ir5ValueERKN5slang3ast10ExpressionE+0x675): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Expression const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::inside_comparison(slang_frontend::EvalContext&, ir::Value, slang::ast::Expression const&):(.text._ZN14slang_frontend17inside_comparisonERNS_11EvalContextEN2ir5ValueERKN5slang3ast10ExpressionE+0x6b5): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Expression const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::inside_comparison(slang_frontend::EvalContext&, ir::Value, slang::ast::Expression const&):(.text._ZN14slang_frontend17inside_comparisonERNS_11EvalContextEN2ir5ValueERKN5slang3ast10ExpressionE+0x6e3): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Expression const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::EvalContext::operator()(slang::ast::Expression const&):(.text._ZN14slang_frontend11EvalContextclERKN5slang3ast10ExpressionE+0x5c): error: undefined reference to 'slang_frontend::transfer_attrs(slang_frontend::NetlistContext&, slang::ast::Expression const&, slang_frontend::AttributeGuard&)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::EvalContext::operator()(slang::ast::Expression const&):(.text._ZN14slang_frontend11EvalContextclERKN5slang3ast10ExpressionE+0x2de2): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Symbol const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::EvalContext::operator()(slang::ast::Expression const&):(.text._ZN14slang_frontend11EvalContextclERKN5slang3ast10ExpressionE+0x2e10): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Symbol const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::NetlistContext::wire(slang::ast::Symbol const&):(.text._ZN14slang_frontend14NetlistContext4wireERKN5slang3ast6SymbolE+0x13a): error: undefined reference to 'slang_frontend::wire_missing_(slang_frontend::NetlistContext&, slang::ast::Symbol const&, char const*, int)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::NetlistContext::add_wire(slang::ast::ValueSymbol const&):(.text._ZN14slang_frontend14NetlistContext8add_wireERKN5slang3ast11ValueSymbolE+0x3e): error: undefined reference to 'slang_frontend::transfer_attrs(slang_frontend::NetlistContext&, slang::ast::Symbol const&, slang_frontend::AttributeGuard&)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::StatementExecutor::handle(slang::ast::BlockStatement const&):(.text._ZN14slang_frontend17StatementExecutor6handleERKN5slang3ast14BlockStatementE+0x9c): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Statement const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::StatementExecutor::handle(slang::ast::ReturnStatement const&):(.text._ZN14slang_frontend17StatementExecutor6handleERKN5slang3ast15ReturnStatementE+0x188): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Statement const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::StatementExecutor::handle(slang::ast::ConditionalStatement const&):(.text._ZN14slang_frontend17StatementExecutor6handleERKN5slang3ast20ConditionalStatementE+0x449): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Statement const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::StatementExecutor::handle(slang::ast::ConditionalStatement const&):(.text._ZN14slang_frontend17StatementExecutor6handleERKN5slang3ast20ConditionalStatementE+0x477): error: undefined reference to 'slang_frontend::unimplemented_(slang::ast::Statement const&, char const*, int, char const*)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/slang_frontend.o:slang_frontend.cc:function slang_frontend::PopulateNetlist::handle_ff_process(slang::ast::ProceduralBlockSymbol const&, slang::ast::SignalEventControl const&, slang::ast::StatementBlockSymbol const*, std::vector<slang::ast::Statement const*, std::allocator<slang::ast::Statement const*> >, slang::ast::Statement const&, std::span<slang_frontend::TimingPatternInterpretor::AsyncBranch, 18446744073709551615ul>):(.text._ZN14slang_frontend15PopulateNetlist17handle_ff_processERKN5slang3ast21ProceduralBlockSymbolERKNS2_18SignalEventControlEPKNS2_20StatementBlockSymbolESt6vectorIPKNS2_9StatementESaISF_EERSE_St4spanINS_24TimingPatternInterpretor11AsyncBranchELm18446744073709551615EE+0xf7f): error: undefined reference to 'slang_frontend::transfer_attrs(slang_frontend::NetlistContext&, slang::ast::Symbol const&, slang_frontend::AttributeGuard&)'
bazel-out/k8-opt-exec-ST-d57f47055a04/bin/third-party/slang-elab/src/_objs/slang_frontend/addressing.o:addressing.cc:function slang_frontend::sign_extend(ir::Value const&, unsigned long):(.text._ZN14slang_frontendL11sign_extendERKN2ir5ValueEm+0x4c): error: undefined reference to 'ir::Net::repeat(unsigned long)'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/src/syn/src/elab/BUILD
+++ b/src/syn/src/elab/BUILD
@@ -50,4 +50,5 @@ cc_library(
         "@fmt",
         "@sv-lang//:libsvlang",
     ],
+    alwayslink = True,
 )

